### PR TITLE
Extract TestCase class

### PR DESCRIPTION
**Pull request by mkaput**
cc #625

### CODE REVIEW

File: protostar/commands/test/test_suite.py [hunk 1]
```diff
@@ -3,8 +3,16 @@
...
+
 @dataclass(frozen=True)
 class TestSuite:
     test_path: Path
-    test_case_names: List[str]
+    test_cases: List[TestCase]
     setup_fn_name: Optional[str] = None
+
+    def get_test_case_names(self) -> List[str]:
+        return [tc.test_fn_name for tc in self.test_cases]
```
Comment (kasperski95) [inline protostar/commands/test/test_suite.py:18]:
```suggestion
    @property
    def test_case_names(self) -> List[str]:
        return [tc.test_fn_name for tc in self.test_cases]
```

**Review by kasperski95 — APPROVED** [review]

File: protostar/commands/test/test_collector.py [hunk 2]
```diff
@@ -57,31 +57,24 @@ class TestSuiteInfo:
...
+    def match_test_case(self, test_case: TestCase) -> bool:
+        name = test_case.test_fn_name
+        return fnmatch_any(name, self.test_case_globs) and (
+            not fnmatch_any(name, self.ignored_test_case_globs)
+        )
+
+
+def fnmatch_any(name: str, pats: Iterable[str]) -> bool:
+    for pat in pats:
+        if fnmatch(name, pat):
```
[the remark this answers is not in the document]
Comment (mkaput) [reply protostar/commands/test/test_collector.py:74]: short for `pattern`. took argument names from `fnmatch` signature to be aligned

File: protostar/commands/test/test_suite.py (hunk 1 above)
Comment (mkaput) [reply protostar/commands/test/test_suite.py:18]: this is non-trivial operation (its constructing a new list each call), hence I used a method approach

Comment (mkaput) [reply protostar/commands/test/test_suite.py:18]: maybe `collect_test_case_names`?

Comment (mkaput) [reply protostar/commands/test/test_suite.py:18]: Done.

File: protostar/commands/test/test_collector.py [hunk 3]
```diff
@@ -57,31 +57,24 @@ class TestSuiteInfo:
...
-        return list(result)
-
-    def _find_matching_any_test_case_glob(self, test_case_names: List[str]) -> Set[str]:
-        result: Set[str] = set()
-        for test_case_name in test_case_names:
-            for test_case_glob in self.test_case_globs:
-                if fnmatch(test_case_name, test_case_glob):
-                    result.add(test_case_name)
-        return result
+    def match_test_cases(self, test_cases: Iterable[TestCase]) -> Iterable[TestCase]:
```
[the remark this answers is not in the document]
Comment (mkaput) [reply protostar/commands/test/test_collector.py:60]: Done. `filter_test_cases`

File: protostar/commands/test/test_collector.py [hunk 4]
```diff
@@ -57,31 +57,24 @@ class TestSuiteInfo:
...
-        result = (
-            test_case_names.copy()
-        )  # copy prevents changing lengths of this collection during loop execution
-        for test_case_name in test_case_names:
-            for ignored_test_case_glob in self.ignored_test_case_globs:
-                if fnmatch(test_case_name, ignored_test_case_glob):
-                    result.remove(test_case_name)
-                    break
-        return result
+    def match_test_case(self, test_case: TestCase) -> bool:
```
[the remark this answers is not in the document]
Comment (mkaput) [reply protostar/commands/test/test_collector.py:65]: Done. `is_test_case_included`

File: protostar/commands/test/test_collector.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (kasperski95) [reply protostar/commands/test/test_collector.py:74]: `fnmatch` signature sucks then. Anyway resolving.

**Pull request merged**